google-services/Location/BasicLocationSample/BasicLocationSample/MainActivity.cs

```diff
@@ -1,95 +1,184 @@
+
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+            SetContentView(Resource.Layout.main_activity);
+
+            mLatitudeLabel = GetString(Resource.String.latitude_label);
+            mLongitudeLabel = GetString(Resource.String.longitude_label);
+            mLatitudeText = FindViewById<TextView>(Resource.Id.latitude_text);
+            mLongitudeText = FindViewById<TextView>(Resource.Id.longitude_text);
+
+            mFusedLocationClient = LocationServices.GetFusedLocationProviderClient(this);
+        }
+
+        protected override void OnStart()
+        {
+            base.OnStart();
+
+            if (!CheckPermissions())
+            {
+                RequestPermissions();
+            }
+            else
+            {
+                GetLastLocation();
+            }
+        }
+
+        private void GetLastLocation()
+        {
+            mFusedLocationClient.LastLocation.AddOnCompleteListener(new OnCompleteListener { Activity = this });
+        }
+
+        public void ShowSnackbar(string text)
+        {
+            View container = FindViewById<TextView>(Resource.Id.main_activity_container);
+            if (container != null)
+            {
+                Snackbar.Make(container, text, Snackbar.LengthLong).Show();
+            }
+        }
+
+        public void ShowSnackbar(int mainTextStringId, int actionStringId, IOnClickListener listener)
+        {
+            Snackbar.Make(FindViewById(Resource.Id.wrap_content), GetString(mainTextStringId), Snackbar.LengthIndefinite)
+                .SetAction(GetString(actionStringId), listener).Show();
+        }
+
+        private bool CheckPermissions()
+        {
+            var permissionState = ContextCompat.CheckSelfPermission(this, Manifest.Permission.AccessCoarseLocation);
+            return permissionState == (int)Permission.Granted;
+        }
+
+        public void StartLocationPermissionRequest()
+        {
+            ActivityCompat.RequestPermissions(this, new[] { Manifest.Permission.AccessCoarseLocation }, RequestPermissionsRequestCode);
+        }
+
+        private void RequestPermissions()
+        {
+            bool shouldProvideRationale = ActivityCompat.ShouldShowRequestPermissionRationale(this, Manifest.Permission.AccessCoarseLocation);
+
+            if (shouldProvideRationale)
+            {
+                Log.Info(Tag, "Displaying permission rationale to provide additional context.");
+                ShowSnackbar(Resource.String.permission_rationale, Android.Resource.String.Ok, new RequestPermissionsClickListener { Activity = this });
+            }
+            else
+            {
+                Log.Info(Tag, "Requesting permission");
+                StartLocationPermissionRequest();
+            }
+        }
+
+        public void OnRequestPermissionsResult(int requestCode, string[] permissions, int[] grantResults)
```

Comment:
This should override the base `OnRequestPermissionsResult ` method.

Reply:
Updated.